dashboard/lib/projects_list.rb

```diff
@@ -17,6 +17,22 @@ module ProjectsList
   }.freeze
 
   class << self
+    # Look up every project associated with the provided user_id, which is not hidden or deleted.
+    # Return a set of metadata which can be used to display a table of personal projects in the UI.
+    # @param user_id
+    # @return [Array<Hash>] An array with each entry representing a project.
+    def fetch_personal_projects(user_id)
+      personal_projects_list = []
+      storage_id = storage_id_for_user_id(user_id)
+      PEGASUS_DB[:storage_apps].where(storage_id: storage_id, state: 'active').each do |project|
+        channel_id = storage_encrypt_channel_id(storage_id, project[:id])
+        project_data = get_personal_project_row_data(project, channel_id)
+        project_data["publishedAt"] = project[:published_at] if project_data
+        personal_projects_list << project_data if project_data
```

Comment:
I think it introduces a little extra complexity to allow `get_personal_project_row_data` to return nil.  After all, those projects exist, they are just hidden.  What if we included `hidden` in the returned information and then filtered those out here?

```rb
def fetch_personal_projects(user_id)
  storage_id = storage_id_for_user_id user_id
  personal_projects = PEGASUS_DB[:storage_apps].
    where(storage_id: storage_id, state: 'active').
    map do |project|
      channel_id = storage_encrypt_channel_id storage_id, project[:id]
      get_personal_project_row_data project, channel_id
    end
  personal_projects.reject {|project| project[:hidden]}
end
```